strings/base_activation.h

```diff
@@ -400,10 +400,10 @@ namespace winrt::impl
         return factory.call(static_cast<CastType>(callback));
     }
 
-    template <typename Class, typename Interface = Windows::Foundation::IActivationFactory>
-    com_ref<Interface> try_get_activation_factory(hresult_error* exception = nullptr) noexcept
+    template <typename Interface = Windows::Foundation::IActivationFactory>
+    com_ref<Interface> try_get_activation_factory(std::wstring_view className, hresult_error* exception = nullptr) noexcept
```

Comment:
This should be a `param::hstring const& name` parameter - this is the "string view" type we use internally for improved performance.